pinot-core/src/main/java/org/apache/pinot/core/query/request/context/FilterInfo.java

```diff
@@ -0,0 +1,116 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.query.request.context;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import org.apache.pinot.core.query.request.context.predicate.Predicate;
+
+
+/**
+ * The {@code FilterInfo} class encapsulates the information of the filter in the query. Both WHERE clause and HAVING
+ * clause are modeled as a filter.
+ */
+public class FilterInfo {
```

Comment:
Today execution engine is based on FilterQueryTree. So the wire request contains FilterQuery and FilterSubqueryMap. On the server, FilterPlanNode converts these into FilterQueryTree for each segment level execution.

Why can't we continue to leverage FilterQueryTree? We should just create it once (since it is segment independent) and stash it away in QueryContext. It already encapsulates all the information that this class seems to provide.

Reply:
`FilterQueryTree` will be replaced with the `FilterInfo` in the following PRs of deprecating BrokerRequest.
We want to replace `FilterQueryTree` for the following reasons:
- It has the Thrift object `FilterOperator`
- The `value` is not a good abstraction for Predicate (e.g. for RANGE, it is a single string of `[a\t\tb]`)
- The predicate is created on per segment basis